pkg/controller/v1beta1/inferenceservice/components/predictor.go

```diff
@@ -91,15 +94,27 @@ func (p *Predictor) Reconcile(isvc *v1beta1.InferenceService) error {
 		if isvc.Spec.Predictor.Model.Runtime != nil {
 			r, err := isvcutils.GetServingRuntime(p.client, *isvc.Spec.Predictor.Model.Runtime, isvc.Namespace)
 			if err != nil {
+				isvc.Status.UpdateModelTransitionStatus(v1beta1.InvalidSpec, &v1beta1.FailureInfo{
+					Reason:  v1beta1.RuntimeNotRecognized,
+					Message: "Waiting for runtime to become available",
+				})
 				return err
 			}
 
 			if r.IsDisabled() {
+				isvc.Status.UpdateModelTransitionStatus(v1beta1.InvalidSpec, &v1beta1.FailureInfo{
+					Reason:  v1beta1.RuntimeUnhealthy,
```

Comment:
We might need to define the reason for Disabled runtime instead of `RuntimeUnhealthy`

Reply:
@yuzisun Added a new reason `RuntimeDisabled`.